AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -0,0 +1,228 @@
+
+@registered_plugin
+class PolicePlugin(AbstractPlugin):
+    def __init__(self):
+        super().__init__("PolicePlugin")
+
+        self.html_ids = {
+            "Rank": self.identifier + "_rank",
+            "Rank Name": self.identifier + "_name"
+        }
+
+        self.config_exports = [
+            ConfigExport(
+                "PolicePlugin_set_ranks",
+                "Police -> Set Ranks",
+                self.ask_set_ranks,
+                self.answer_set_ranks
+            )
+        ]
+
+        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+
+    def ask_set_ranks(self):  # TODO Make Ctrl-C work - not sure why it isn't
+        return [
+            Label("Rename different ranks of police to fit the game theme"),
+            Label("Ranks affect position on the police page, and can be used to reward/punish/just have a bit of fun"),
+            InputWithDropDown(
+                title="Select rank to rename",
+                identifier=self.html_ids["Rank"],
+                options=[str(i[0]) + ': ' + i[1] for i in list(GENERIC_STATE_DATABASE.arb_state.get(self.identifier).items())]
+            ),
+            NamedSmallTextbox(
+                title="Enter new name",
+                identifier=self.html_ids["Rank Name"],
+            )
+        ]
+
+    def answer_set_ranks(self, htmlResponse):
+        rank = htmlResponse[self.html_ids['Rank']]
+        new_name = htmlResponse[self.html_ids['Rank Name']]
+        if new_name:
+            GENERIC_STATE_DATABASE.arb_state[self.identifier][rank[0]] = new_name
+            return [Label(f"[POLICE] Successfully renamed rank {rank} to {new_name}")]
+        else:
+            return [Label(f"[POLICE] Rename failed - no name given")]
+
+    def on_event_request_create(self) -> List[HTMLComponent]:  # TODO Make a dependent selector, ideally with police filtering
+        return [
+            Dependency(
+                dependentOn="CorePlugin_assassin_pseudonym",
+                htmlComponents=[
+                    AssassinDependentTextEntry(
+                        pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
+                        identifier=self.html_ids["Rank"],
+                        title="Change police rank? Must be a number from 0-9",
+                        default={}
+                    )
+                ]
+            )
+        ]
+
+    def on_event_create(self, e: Event, htmlResponse) -> List[HTMLComponent]:
+        e.pluginState.setdefault(self.identifier, {})["rank"] = htmlResponse[self.html_ids["Rank"]]
+        return [Label("[POLICE] Success!")]
+
+    def on_event_request_update(self, e: Event) -> List[HTMLComponent]:  # TODO Make a dependent selector, ideally with police filtering
+        return [
+            Dependency(
+                dependentOn="CorePlugin_assassin_pseudonym",
+                htmlComponents=[
+                    AssassinDependentTextEntry(
```

Comment:
[AssassinDependentIntegerEntry](https://github.com/jsyiek/AU2/blob/master/AU2/html_components/AssassinDependentIntegerEntry.py)

Reply:
I really should just have an AssassinDependentEntry factory that takes one of a few pre-defined validators to avoid having to make a new AssassinDependentEntry component for different validators, but oh well. I'll refactor it if I ever have time/if it's ever worth it.